nixos/modules/security/hidepid.nix

```diff
@@ -33,6 +33,12 @@ in
   };
 
   config = mkIf cfg.enable {
+    assertions =
+     [ { assertion = cfg.procGroup != null -> (config.users.groups."${cfg.procGroup}" or "") != "";
+         message = "the group named at security.hideProcessInformation.procGroup does not exist!";
+       }
+     ];
+
```

Comment:
I actually don't think it would be an issue. We already allocate separate GIDs for all services (some even have two and maybe more!) -- added group dedicated to this shouldn't be a problem and would cut much of the complexity.